src/cache.test.js

```diff
@@ -0,0 +1,6 @@
+const { withCacheAsync } = require("./cache");
+
+test("async cache", async () => {
+  const foo = withCacheAsync((x) => x*x)
+  expect(await foo(5)).toEqual(25);
```

Comment:
this is not really testing the cache right ? maybe better to return the timestamp and then call it twice to see it doesn't change

Reply:
Right, using Math.random now